compose/cli/main.py

```diff
@@ -720,6 +724,9 @@ def up(self, project, options):
             raise UserError("--abort-on-container-exit and -d cannot be combined.")
 
         with up_shutdown_context(project, service_names, timeout, detached):
+            # start the event stream first so we don't lose any events
+            event_stream = project.events()
```

Comment:
hmm, I think this is a lie. The generator isn't going to start running until we actually try to read one event from it.  This might need more work.